src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -1107,13 +1107,26 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
   private[service] def processClusterImages(clusterRequest: ClusterRequest): Set[ClusterImage] = {
     val now = Instant.now
 
+    //If welder is enabled for this cluster, we need to ensure that an image is chosen.
+    //We will use the client-supplied image, if present, otherwise we will use a default.
+    //If welder is not enabled, we won't use any image.
+    //Eventually welder will be enabled for all clusters and this will be way cleaner.
+    val welderImage = if(clusterRequest.enableWelder.getOrElse(false)) {
+      val i = clusterRequest.welderDockerImage.getOrElse(dataprocConfig.welderDockerImage)
+      Some(ClusterImage(Welder, i, now))
+    } else None
+
     val images = Set(
```

Comment:
I think this logic has a bug. If the user passes enableWelder == true, but doesn't pass in any images, they will wind up with a welder and proxy image, but no Jupyter.

Reply:
I'm making a fix

Reply:
https://github.com/DataBiosphere/leonardo/pull/980